### PR TITLE
Port data quality checks from solarforecastarbiter

### PR DESCRIPTION
**Pull request by wfvining**
Adds data-quality checks from [solarforecastarbiter](https://github.com/SolarArbiter/solarforecastarbiter-core) to the RdTools filtering module. There are two main features added in this PR:
* filtering stale data (periods where the value has not changed) and interpolated data (looks for periods of linear interpolation)
* Validation of irradiance observations using the QCRad algorithm

### CODE REVIEW

File: rdtools/filtering.py [hunk 1]
```diff
@@ -92,3 +98,188 @@ def csi_filter(measured_poa, clearsky_poa, threshold=0.15):
...
+        ValueError if window < 3
+    '''
+    if window < 3:
+        raise ValueError('window set to {}, must be at least 3'.format(window))
+
+    # reduce window by 1 because we're passing the first difference
+    flags = stale_values_filter(data.diff(periods=1), window=window-1, rtol=rtol,
+                                atol=atol)
+    return flags
+
```
Comment (cwhanse) [inline rdtools/filtering.py]: PEP8 - two blank lines between functions

File: rdtools/qcrad.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
+'''Implementation of the QCRad algorithm for validating irradiance
+obseervations.
```
Comment (cwhanse) [inline rdtools/qcrad.py]:
```suggestion
observations.
```

File: rdtools/qcrad.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+@author: Cliff Hansen
```
Comment (cwhanse) [inline rdtools/qcrad.py]:
```suggestion

```

File: rdtools/qcrad.py [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+    Returns
+    -------
+    ghi_limit_flag : Series or None, default None
+        True if value passes physically-possible test
+    dhi_limit_flag : Series or None, default None
+    dhi_limit_flag : Series or None, default None
+
+    References
+    ----------
+    [1] C. N. Long and Y. Shi, An Automated Quality Assessment and Control
```
Comment (cwhanse) [inline rdtools/qcrad.py]:
```suggestion
    .. [1] C. N. Long and Y. Shi, An Automated Quality Assessment and Control
```

File: rdtools/qcrad.py [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+        Algorithm for Surface Radiation Measurements, The Open Atmospheric
```
Comment (cwhanse) [inline rdtools/qcrad.py]: line up other text with [ in line above

File: rdtools/qcrad.py [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+    dni_limit_flag = _check_limits(dni, limits['dni_lb'], dni_ub)
+    dni_limit_flag.name = 'dni_limit_flag'
+
+    return dni_limit_flag
+
+
+def check_irradiance_limits(solar_zenith, dni_extra, ghi=None, dhi=None,
+                            dni=None, limits=None):
+    '''
+    Tests for physical limits on GHI, DHI or DNI using the QCRad criteria.
```
Comment (cwhanse) [inline rdtools/qcrad.py]:
```suggestion
    Tests for physical limits on GHI, DHI or DNI using the QCRad criteria [1]_.
```

File: rdtools/qcrad.py [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+    # for zenith set lb_ge to handle edge cases, e.g., zenith=0
+    return ((_check_limits(sza, lb=sza_lb, ub=sza_ub, lb_ge=True)) &
+            (_check_limits(ghi, lb=ghi_lb, ub=ghi_ub)) &
+            (_check_limits(ratio, lb=ratio_lb, ub=ratio_ub)))
+
+
+def check_irradiance_consistency(ghi, solar_zenith, dni_extra, dhi, dni,
+                                 param=None):
+    '''
+    Checks consistency of GHI, DHI and DNI. Not valid for night time.
```
Comment (cwhanse) [inline rdtools/qcrad.py]:
```suggestion
    Checks consistency of GHI, DHI and DNI using the QCRad algorithm [1]_. Not valid for night time.
```

File: rdtools/qcrad.py [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+    Returns
+    -------
+    consistent_components : Series
+        True if ghi, dhi and dni components are consistent.
+    diffuse_ratio_limit : Series
+        True if diffuse to ghi ratio passes limit test.
+
+    References
+    ----------
+    [1] C. N. Long and Y. Shi, An Automated Quality Assessment and Control
```
Comment (cwhanse) [inline rdtools/qcrad.py]:
```suggestion
    .. [1] C. N. Long and Y. Shi, An Automated Quality Assessment and Control
```
